### ISSUE TITLE
ElevenLabs download audio in WebGL Build

### ISSUE THREAD

**Issue by SebastianBlandon** [RageAgainstThePixel/com.rest.elevenlabs#48]
# Bug Report

## Overview
I have a problem with downloading from ElevenLabs of the audios in a WebGl build of my application, follow the browser logs is by CORS, that I can not load the audios and failure to download with Utilities.WebRequestRest. Has anyone encountered this error?

## Screenshots
<img width="1805" alt="Screenshot 2023-11-22 at 11 00 56 AM" src="https://github.com/RageAgainstThePixel/com.rest.elevenlabs/assets/45244286/bbe22cf6-be1b-4dbe-8c03-106e66eafef9">

**Comment by StephenHodgson** [comment]
Please replace screenshot with error text

**Comment by SebastianBlandon** [comment]
#List errors:

1. Cross origin requests are only supported for HTTP.
2. Fetch API cannot load file:///tmp/download_cache/ElevenLabs/TextToSpeech/onwK4e9ZLuTAKqWW03F9/cF3SDAlKxqH9JJIxDtRe.ogg due to access control checks.
3. Utilities.WebRequestRest.RestException: Failed to download audio clip from "file:///tmp/download_cache/ElevenLabs/TextToSpeech/onwK4e9ZLuTAKqWW03F9/cF3SDAlKxqH9JJIxDtRe.ogg"!
  at Utilities.WebRequestRest.Rest.DownloadAudioClipAsync (System.String url, UnityEngine.AudioType audioType, System.String fileName, Utilities.WebRequestRest.RestParameters parameters, System.Threading.CancellationToken cancellationToken) [0x00000] in <00000000000000000000000000000000>:0

**Comment by StephenHodgson** [comment]
I think there is an additional step you must to to enable CORS for your webgl build.

But also, I think the download cache works differently on WebGL since there isn't access to local disk.

**Comment by SebastianBlandon** [comment]
I'm looking for documentation on this but I can't find anything that works for me, do you have any idea if this is set in the player configuration?

**Comment by StephenHodgson** [comment]
I think it might actually be a bug with my [com.utilities.rest](https://github.com/RageAgainstThePixel/com.utilities.rest) library.

I'll do some validations and get back to you

**Comment by StephenHodgson** [comment]
https://github.com/RageAgainstThePixel/com.utilities.rest/issues/51

**Comment by StephenHodgson** [comment]
Due to CORS policy of image storage in local WebGL builds you will get the generated image's URL however it will not be downloaded using UnityWebRequest until you run it out of localhost, on a server.

**Comment by StephenHodgson** [comment]
Yeah for now this package does not support WebGL. There's too many places where I'm attempting to cache them and save them to disk.

**Comment by yosun** [comment]
eh, any update for webgl support

**Comment by StephenHodgson** [comment]
https://github.com/RageAgainstThePixel/com.utilities.audio/pull/40

**Comment by StephenHodgson** [comment]
caching has already been made optional, and with the audio utils update we should be able to stream audio in webgl after 3.5.0
